src/overlays/actors/ovl_En_Bigpamet/z_en_bigpamet.c

```diff
@@ -90,121 +113,763 @@ static DamageTable D_80A29734 = {
+    func_800B9010(&this->actor, NA_SE_EN_B_PAMET_ROLL - SFX_FLAG);
+    this->unk_29E++;
+    this->unk_29E = CLAMP_MAX(this->unk_29E, 20);
+    if (this->collider.base.atFlags & AT_HIT) {
+        this->collider.base.atFlags &= ~AT_HIT;
+        if ((s16)(this->actor.yawTowardsPlayer - this->actor.world.rot.y) > 0) {
+            this->actor.world.rot.y = this->actor.world.rot.y - 0x300;
+        } else {
+            this->actor.world.rot.y = this->actor.world.rot.y + 0x300;
+        }
+    }
+
+    if (play->gameplayFrames & 1) {
+        func_80A27FE8(this, play);
+    }
+
+    if (this->actor.bgCheckFlags & 8) {
+        quake = Quake_Add(play->cameraPtrs[play->activeCamera], 3);
+        this->actor.velocity.y = this->unk_29E * 0.375f;
+        Quake_SetSpeed(quake, 0x4E20);
+        Quake_SetQuakeValues(quake, 15, 0, 0, 0);
+        Quake_SetCountdown(quake, 10);
+        Rumble_Request(this->actor.xyzDistToPlayerSq, 180, 20, 100);
+        func_80A27B58(this);
+        func_80A27DD8(this, play);
+        func_80A28970(this);
+    }
+}
+
+void func_80A28970(EnBigpamet* this) {
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_PAMET_CUTTER_OFF);
+    this->actor.shape.rot.z = 0;
+    this->collider.base.atFlags &= ~AT_ON;
+    this->collider.info.bumper.dmgFlags = 0xF7CFFFFF;
+    this->actor.speedXZ = 0.0f;
+    this->actionFunc = func_80A289C8;
+}
+
+void func_80A289C8(EnBigpamet* this, PlayState* play) {
+    this->unk_2A8 -= 0.1f;
+    if (this->unk_2A8 < 0.5f) {
+        func_80A28A28(this);
+    }
+    func_80A27DD8(this, play);
+}
+
+void func_80A28A28(EnBigpamet* this) {
+    this->actor.draw = EnBigpamet_Draw;
+    Animation_MorphToPlayOnce(&this->skelAnime1, &object_tl_Anim_0031DC, -3.0f);
+    this->unk_2A8 = 0.1f;
+    this->unk_2A4 = 1.0f;
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->actionFunc = func_80A28A98;
+}
+
+void func_80A28A98(EnBigpamet* this, PlayState* play) {
+    f32 curFrame;
+
+    Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0x600);
+
+    if (SkelAnime_Update(&this->skelAnime1)) {
+        if ((this->actor.parent->params == 7) || (this->actor.parent->params == 8)) {
+            func_80A283F0(this);
+        } else {
+            func_80A284E4(this);
+        }
+    } else {
+        curFrame = this->skelAnime1.curFrame;
+
+        if (curFrame > 7.0f) {
+            this->unk_2A8 = 1.5f - ((curFrame - 7.0f) * 0.16666667f);
+            this->unk_2A4 = 1.5f - ((curFrame - 7.0f) * 0.16666667f);
+        } else {
+            this->unk_2A8 = (0.2f * curFrame) + 0.1f;
+            this->unk_2A4 = (0.071428575f * curFrame) + 1.0f;
+        }
+    }
+}
+
+void func_80A28B98(EnBigpamet* this, PlayState* play) {
+    Actor* sp24;
+
+    Animation_PlayLoop(&this->skelAnime1, &object_tl_Anim_000AF4);
+
+    this->collider.base.atFlags &= ~AT_ON;
+    this->collider.info.bumper.dmgFlags = 0xF7CFFFFF;
+    this->collider.base.atFlags &= ~(2 | 4);
```

Comment:
```suggestion
    this->collider.base.atFlags &= ~(AT_HIT | AT_BOUNCED);
```
Hmm, wonder if your script only catches single